.github/workflows/examples_matrix.yml

```diff
@@ -0,0 +1,90 @@
+name: ExamplesMatrix
+
+on:
+  workflow_call:
+    inputs:
+      name:
+        type: string
+        description: "The name of the workflow used for the concurrency group."
+        required: true
+      # examples:
+      #   type: sequence
+      #   description: "The examples to run."
+      #   required: true
+      matrix_linux_command_override:
+        type: string
+        description: "The command of the current Swift version linux matrix job to execute."  
+      matrix_linux_command:
+        type: string
+        description: "The command of the current Swift version linux matrix job to execute."
+        required: true
+      matrix_linux_nightly_6_0_enabled:
+        type: boolean
+        description: "Boolean to enable the nightly 6.0 Swift version matrix job. Defaults to true."
+        default: true
+      matrix_linux_nightly_6_0_container_image:
+        type: string
+        description: "Container image for the nightly 6.0 Swift version matrix job. Defaults to matching Swift Ubuntu image."
+        default: "swiftlang/swift:nightly-6.0-jammy"
+      matrix_linux_nightly_6_0_command_override:
+        type: string
+        description: "The command of the nightly 6.0 Swift version linux matrix job to execute."
+      matrix_linux_nightly_main_enabled:
+        type: boolean
+        description: "Boolean to enable the nightly main Swift version matrix job. Defaults to true."
+        default: true
+      matrix_linux_nightly_main_container_image:
+        type: string
+        description: "Container image for the nightly main Swift version matrix job. Defaults to matching Swift Ubuntu image."
+        default: "swiftlang/swift:nightly-main-jammy"
+      matrix_linux_nightly_main_command_override:
+        type: string
+        description: "The command of the nightly main Swift version linux matrix job to execute."
+
+## We are cancelling previously triggered workflow runs
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ inputs.name }}
+  cancel-in-progress: true
+ 
+jobs:
+  linux:
+    name: Linux (${{ matrix.swift.swift_version }})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # This should be passed as an argument in input. Can we pass arrays as argument ?
+        examples : [ "HelloWorld", "APIGateway" ]
+        # examples: ${{ inputs.examples }}
+        
+        # We are specifying only the major and minor of the docker images to automatically pick up the latest patch release
+        swift:
```

Comment:
I honestly think that testing with a single swift version (likely the latest released should be sufficient)

Reply:
I think it's good to test on the 6.x branch and the main branch.
But I will disagree and commit on this.